public/app/features/library-panels/components/LibraryPanelsSearch/LibraryPanelsSearch.test.tsx

```diff
@@ -319,4 +321,54 @@ describe('LibraryPanelsSearch', () => {
       expect(within(card()).getByLabelText(/delete button on panel type card/i)).toBeInTheDocument();
     });
   });
+
+  describe('when mounted with showSecondaryActions and a specific folder', () => {
+    describe('and user deletes a panel', () => {
+      it('should call api with correct params', async () => {
+        const { getLibraryPanelsSpy } = await getTestContext(
+          { showSecondaryActions: true, currentFolderUID: 'wfTJJL5Wz' },
+          {
+            elements: [
+              {
+                name: 'Library Panel Name',
+                uid: 'uid',
+                description: 'Library Panel Description',
+                folderUid: 'wfTJJL5Wz',
+                model: { type: 'timeseries', title: 'A title' } as Panel,
+                type: 'timeseries',
+                version: 1,
+                meta: {
+                  folderName: 'General',
+                  folderUid: '',
+                  connectedDashboards: 0,
+                  created: '2021-01-01 12:00:00',
+                  createdBy: { id: 1, name: 'Admin', avatarUrl: '' },
+                  updated: '2021-01-01 12:00:00',
+                  updatedBy: { id: 1, name: 'Admin', avatarUrl: '' },
+                },
+              },
+            ],
+            perPage: 40,
+            page: 1,
+            totalCount: 1,
+          }
+        );
+
+        await userEvent.click(screen.getByLabelText(/delete button on panel type card/i));
+        await waitFor(() => expect(screen.getByText('Do you want to delete this panel?')).toBeInTheDocument());
+        await userEvent.click(screen.getByRole('button', { name: 'Delete' }));
+
+        await waitFor(() => {
+          expect(getLibraryPanelsSpy).toHaveBeenCalledWith({
+            searchString: '',
+            folderFilterUIDs: ['wfTJJL5Wz'],
+            page: 1,
+            typeFilter: [],
+            sortDirection: undefined,
+            perPage: 40,
+          });
```

Comment:
Does this test actually reproduce the issue, since it only relates to one folder? Or is this a test scenario more suited for e2e?

Reply:
It doesn't reproduce the issue raised exactly, instead it reproduces the problem that there was behind the scenes, which was when we deleted the library panel, it was calling the `getLibraryPanels` with the wrong information (missing folder id). We could create an e2e test to test the full bug though! I am still not sure what our policy is around that, since I know e2e tests can be slower, do we think that adding one for this scenario would be valuable?

Reply:
Could you add another library panel in another folder, and assert that it hasn't been called/it doesn't appear too? Then I would be fine without an e2e test. I am not sure how much e2e coverage we have for nested folders in general, maybe something worth looking into.

Reply:
We are mocking `getLibraryPanels` though, so we don't actually know what it would return in reality 😬  so its hard for us with a test to figure out what would be shown after this call since we are the ones faking the results...

I don't think this bug is specific to nested folders btw, it is something that happened before as well

Reply:
Yes, I do remember this happening with nested folders. Maybe an e2e test is the way to go then.